app/Http/Controllers/CompetitionsController.php

```diff
@@ -248,4 +250,72 @@ public function updateFeaturedReportback(FeaturedReportbackRequest $request, Com
 
         return redirect()->route('competitions.message', [$competition, $competition->contest])->with('status', 'Featured reportback has been updated!');
     }
+
+    /**
+     * Get the leaderboard photos form.
+     *
+     * @param  \Gladiator\Models\Competition  $competition
+     * @param  \Gladiator\Models\Message  $message
+     * @return \Illuminate\Http\Response
+     */
+    public function editLeaderboardPhotos(Competition $competition, Message $message)
+    {
+        $key = generate_model_flash_session_key($competition, ['includeActivity' => true]);
+        if (session()->has($key)) {
+            $competition = session($key);
+            session()->reflash();
+        } else {
+            $competition = $this->manager->getCompetitionOverview($competition, true);
+        }
+
+        $leaderboard = $competition->activity['active'];
+        $topThree = $this->manager->getTopThreeReportbacks($leaderboard, ['includeUserIds' => true]);
+
+        $photos = [];
+
+        foreach ($topThree as $key => $user) {
+            $photos[] = LeaderboardPhotos::where('competition_id', '=', $competition->id)->where('message_id', '=', $message->id)->where('user_id', '=', $user['user_id'])->first();
+        }
+
+        return view('competitions.leaderboard_photos.edit', compact('competition', 'message', 'photos', 'topThree'));
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Gladiator\Requests\LeaderboardPhotosRequest  $request
+     * @param  \Gladiator\Models\Competition  $competition
+     * @param  \Gladiator\Models\Message  $message
+     * @return \Illuminate\Http\Response
+     */
+    public function updateLeaderboardPhotos(LeaderboardPhotosRequest $request, Competition $competition, Message $message)
+    {
+        foreach (range(0, 2) as $i) {
```

Comment:
interesting. Could you just use a regular `for` statement here, othewise it reads as a while loop, which is not what is happening. So:

`for ($i = 0; $i <= 2; $i++) {}` and then check that the `$reportback_item_id` is not null before setting the photo.

Want to shoot for readability here in case another developer jumps on this app.